core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -664,113 +671,229 @@ private void processFailure(SearchContext context, Throwable t) {
         }
     }
 
-    private void parseTemplate(ShardSearchRequest request, SearchContext searchContext) {
-
-        BytesReference processedQuery;
-        if (request.template() != null) {
-            ExecutableScript executable = this.scriptService.executable(request.template(), ScriptContext.Standard.SEARCH, searchContext);
-            processedQuery = (BytesReference) executable.run();
-        } else {
-            if (!hasLength(request.templateSource())) {
+    private void parseSource(SearchContext context, SearchSourceBuilder source) throws SearchParseException {
+        // nothing to parse...
+        if (source == null) {
                 return;
             }
-            XContentParser parser = null;
-            Template template = null;
 
+        context.from(source.from());
+        context.size(source.size());
+        ObjectFloatHashMap<String> indexBoostMap = source.indexBoost();
+        if (indexBoostMap != null) {
+            Float indexBoost = indexBoostMap.get(context.shardTarget().index());
+            if (indexBoost != null) {
+                context.queryBoost(indexBoost);
+            }
+        }
+        if (source.query() != null) {
+            context.parsedQuery(context.queryParserService().parse(source.query()));
```

Comment:
this .parse call confuses me, what does it parse, the `source.query()` is already parsed no?